test/plugins/sqlite.js

```diff
@@ -0,0 +1,102 @@
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+const expect = require("chai").expect;
+const Msg = require("../../src/models/msg");
+const Helper = require("../../src/helper");
+const MessageStorage = require("../../src/plugins/sqlite.js");
+
+describe("SQLite Message Storage", function() {
+	const expectedPath = path.join(Helper.getHomePath(), "logs", "testUser.sqlite3");
+	let store;
+
+	// Delete database file from previous test run
+	before((done) => {
+		store = new MessageStorage();
+
+		if (fs.existsSync(expectedPath)) {
+			fs.unlink(expectedPath, done);
+		} else {
+			done();
+		}
+	});
+
+	it("should resolve an empty array when disabled", (done) => {
+		store.getMessages(null, null).then((messages) => {
+			expect(messages).to.be.empty;
+			done();
+		});
+	});
+
+	it("should create database file", () => {
+		expect(store.isEnabled).to.be.false;
+		expect(fs.existsSync(expectedPath)).to.be.false;
+
+		store.enable("testUser");
+
+		expect(store.isEnabled).to.be.true;
+		expect(fs.existsSync(expectedPath)).to.be.true;
+	});
+
+	it("should create tables", (done) => {
+		store.database.serialize(() =>
+			store.database.all("SELECT name, tbl_name, sql FROM sqlite_master WHERE type = 'table'", (err, row) => {
+				expect(err).to.be.null;
+				expect(row).to.deep.equal([{
+					name: "options",
+					tbl_name: "options",
+					sql: "CREATE TABLE options (name TEXT, value TEXT, CONSTRAINT name_unique UNIQUE (name))",
+				},
+				{
+					name: "messages",
+					tbl_name: "messages",
+					sql: "CREATE TABLE messages (network TEXT, channel TEXT, time INTEGER, type TEXT, msg TEXT)",
+				}]);
+
+				done();
+			})
+		);
+	});
+
+	it("should insert schema version to options table", (done) => {
+		store.database.serialize(() =>
+			store.database.get("SELECT value FROM options WHERE name = 'schema_version'", (err, row) => {
+				expect(err).to.be.null;
+
+				// Should be sqlite.currentSchemaVersion,
+				// compared as string because it's returned as such from the database
+				expect(row.value).to.equal("1");
```

Comment:
I'm expecting this to be directly handled by whatever migration mechanism we end up using, but in the present case, you could make `schema_version` a column of `options`, which wouldn't be key/value anymore but a single row with as many columns as we have options.
Version could then be the expected type. Also slightly easier to deal with default values, etc. Only caveat is to have to ensure only 1 line ever lives there, but that's worth it IMO.
Thoughts?

Reply:
I went with `options` because I was thinking we could expand that table to store other details (hence K/V). Do you want me to change it?

Reply:
I'm undecided on this. I totally understand the need for arbitrary K/V, but also it's just as common to have one column per option (even future ones) as it is to have random K/V/T (type) rows.

I think if we change our mind in a future version, it would be easier to go from 1-row-multi-columns to K/V rows than the opposite, and that also offers native types, which is cool.
If you're not okay with this, at least add a type column maybe?